packages/gatsby-script/src/gatsby-script.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect } from "react"
+import { ReactElement } from "react"
+
+export enum ScriptStrategy {
+  preHydrate = `pre-hydrate`,
+  postHydrate = `post-hydrate`,
+  idle = `idle`,
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export interface ScriptProps {
+  src: string
+  strategy?: ScriptStrategy
+}
+
+export function Script(props: ScriptProps): ReactElement {
+  const { src, strategy = ScriptStrategy.postHydrate } = props || {}
+
+  useEffect(() => {
+    switch (strategy) {
+      case ScriptStrategy.postHydrate:
+        injectScript(src, strategy)
+        break
+      case ScriptStrategy.idle:
+        requestIdleCallback(() => {
+          injectScript(src, strategy)
+        })
+        break
+      default:
+        return
+    }
+  }, [])
```

Comment:
Do we need <script> unmount handler when parent component gets unmounted?

Reply:
Good question, thinking more on this it would match the behavior of the regular `<script>` tag (e.g. if a `<script>` tag's parent node is removed, it's also removed)